### PR TITLE
Add AWS deployment

### PR DESCRIPTION
**Pull request by jack-michaud**
- Adds cloudformation template
- Adds circleci config that deploys the CFT

Edit: Please merge https://github.com/metascouter-io/smash-overlays/pull/13 first

### CODE REVIEW

File: .circleci/cloudformation.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+AWSTemplateFormatVersion: '2010-09-09'
```
Comment (phytoporg) [inline .circleci/cloudformation.yml]: I really ought to try playing with one of these on my own to wrap my head around it.

Would you suggest any particular CF tutorials/references?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          name: Install node dependencies
+          command: yarn install
+      - save_cache:
+          key: deps-{{ .Branch }}-{{ checksum "yarn.lock" }}
+          paths:
+            - "node_modules"
+      - run: 
+          name: Deploy CFT
+          command: aws cloudformation deploy --template .circleci/cloudformation.yml --stack-name smash-overlays --region us-west-1 --no-fail-on-empty-changeset
+          no_output_timeout: 60m
```
Comment (phytoporg) [inline .circleci/config.yml]: 60m seems long?

Comment (jack-michaud) [reply .circleci/config.yml]: Sometimes necessary if we update a cert or CDN. I had to leave it running in a screen session on Pichu for the first deployment

File: .circleci/cloudformation.yml (hunk 1 above)
Comment (jack-michaud) [reply .circleci/cloudformation.yml]: This documents every resource and property https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html

I constantly have 3-4 of these tabs open at a time. Sometimes figuring stuff out concretely involves trial and error (e.g. do I provide the ARN or Name for this property?) Cloudformation is not perfect.

I think my first step would be to deploy an EC2 instance with Cloudformation. It requires you to define a lot of stuff, but it's a good kick-the-tires demo of CFTs

Comment (jack-michaud) [reply .circleci/cloudformation.yml]: https://github.com/awslabs/aws-cloudformation-templates I draw inspiration from these often

**Pull request merged**